flypipe/node.py

```diff
@@ -369,8 +328,49 @@ def _get_consolidated_output_schema(cls, output_schema, output_columns):
             schema = None
         return schema
 
+    def process_transformation_with_cache(
+            self, runnable_node, run_context: RunContext

```

Comment:
Why pass runnable_node instead of doing runnable_node['transformation'].process_transformation_with_cache? Unless you're dealing with a collection of nodes like run_sequential does there should not be a need to do this. It's confusing because it's an instance method on node and yet you're primarily working with another node that is passed in.

Reply:
it needs to read the run status of the node to decide if it reads from cache of if it should run the transformation.

Reply:
Ok I see, I thought you could pass the run status of the node as a parameter but there are a lot of things accessed from the runnable node- status, node_run_context, transformation and output_columns. In that case, I suggest making this a classmethod as the only time the self reference is being used is to call a different classmethod (_get_consolidated_output_schema). 

I would consider this kind of situation a clue that the method is in the wrong place, given most of the things used here really belong to the graph. However I think we decided in the past that we wanted to keep all execution code within the node class so it's probably good to leave it here.